### ISSUE TITLE
Disk throughput reported as 2x of reality

### ISSUE THREAD

**Issue by malventano** [scottchiefbaker/dool#32]
On an NVMe disk with partitions, dool double counts the disk throughput. `/proc/diskstats` for reference:
```
[root@archlinux ~]# cat /proc/diskstats
 259       0 nvme0n1 93309610 876 23882648030 11355661 95197224 17624 24366029343 1398530277 0 28495684 1409885939 0 0 0 0 0 0
 259       1 nvme0n1p1 312 817 7294 32 1 0 1 0 0 27 32 0 0 0 0 0 0
 259       2 nvme0n1p2 19421 59 607630 8511 11571 16968 1498601 39884 0 25094 48396 0 0 0 0 0 0
 259       3 nvme0n1p3 93289488 0 23882019458 11347109 95185652 656 24364530741 1398490393 0 28491730 1409837502 0 0 0 0 0 0
```
Unsure if this also happens with other disk types. The expected behavior would be for disk bandwidth to be based on the root devices only and not doubling with partition IO.

**Comment by exuvo** [comment]
You sure this is not related to https://github.com/scottchiefbaker/dool/issues/13 ?

**Comment by malventano** [comment]
> You sure this is not related to #13 ?

#13 is a bit/byte issue. This issue is 2x, not 8x-10x.

**Comment by scottchiefbaker** [comment]
Is this specific to NVME only?

**Comment by Korkman** [comment]
Can confirm this happens with nvme. I have mdraid, dm-crypt, lvm layered on top. The same stack with sata is not counting twice. I know of other scenarios, but they are hard to impossible to fix. bcache (seems dead anyways, not worth the trouble), zfs zvol.

**Comment by scottchiefbaker** [comment]
Disk stat read is the same for SATA and NVME. I don't know how the numbers would be 2x. We only read what the [kernel is reporting](https://github.com/scottchiefbaker/dool/blob/76049a509e5bfe67c6d19df79f57af68bddd75c3/dool#L784) from `/proc/diskstats`

Is this perhaps a kernel bug?

**Comment by malventano** [comment]
> Is this perhaps a kernel bug?

In my case, it appeared to be reporting nvme0 and nvme0n1p3 combined. The kernel values were correct.

**Comment by scottchiefbaker** [comment]
I think **maybe** I figured this out. The code that reads disk stats is pretty gnarly, so I had to make some guesses.

Try changing the filter in `dool_disk()` to this:

```
self.diskfilter = re.compile('^([hsv]d[a-z]+\d+|cciss/c\d+d\d+p\d+|dm-\d+|md\d+|mmcblk\d+p\d0|VxVM\d+|nvme\d+n\d+p\d+)$')
```

I added the nvme stuff to the end of that list. This SHOULD only read stats from the root `/dev/nvmeXnX` devices. Previously we were adding stats for EVERY partition on each drive to "total".
